client/src/components/PinnedContent/PinnedContent.tsx

```diff
@@ -45,20 +67,32 @@ const get_is_pinned = (
   }
 };
 
-class NonA11yPinnedContent extends React.Component {
-  constructor(props) {
+class NonA11yPinnedContent extends React.Component<
+  NonA11yPinnedContentProps,
+  NonA11yPinnedContentState
+> {
+  placeHolderRef = React.createRef<HTMLDivElement>();
+  contentRef = React.createRef<HTMLDivElement>();
+  static defaultProps: { default_pin_state: boolean };
+
+  constructor(
+    props: NonA11yPinnedContentProps,
+    state: NonA11yPinnedContentState
+  ) {
     super(props);
 
     this.state = {
       is_pinned_local_storage_mirror: get_is_pinned(
         props.local_storage_name,
-        props.default_pin_state
+        props.default_pin_state,
+        state.is_pinned_local_storage_mirror
```

Comment:
In this case, I think it's safe to assume that we can use `null` for the third parameter since it used to default to null in the function (it now requires a boolean or null parameter)